### PR TITLE
check group_ids when making a source

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR checks that group_ids are accessible when making a source.

### CODE REVIEW

File: skyportal/handlers/api/source.py [hunk 1]
```diff
@@ -692,14 +692,19 @@ def post_source(data, user_id, session, refresh_source=True):
...
-            for id in data.pop('group_ids')
-            if int(id) in user_accessible_group_ids
-        ]
-    except KeyError:
+
+    group_ids = data.pop('group_ids')
+    if group_ids is None:
         group_ids = user_group_ids
+    else:
+        group_ids = []
```
Comment (Theodlz) [inline skyportal/handlers/api/source.py:700]: don't think that one works 😉 @mcoughlin

Comment (Theodlz) [reply skyportal/handlers/api/source.py:700]: you overwrite it and loop over it, but well it will always be empty

Comment (mcoughlin) [reply skyportal/handlers/api/source.py:700]: Thanks bud! Should be fixed.

File: skyportal/handlers/api/source.py [hunk 2]
```diff
@@ -692,14 +692,20 @@ def post_source(data, user_id, session, refresh_source=True):
...
-            for id in data.pop('group_ids')
-            if int(id) in user_accessible_group_ids
-        ]
-    except KeyError:
+
+    group_ids = data.pop('group_ids')
+    if group_ids is None:
         group_ids = user_group_ids
+    else:
+        group_ids_tmp = []
```
Comment (Theodlz) [inline skyportal/handlers/api/source.py:700]: Thanks for fixing that! Though I feel like you could even have made it happen without any extra list. All that this does is validate that the ids are accessible to the user and then make sure they are ints. You could have also just made them ints with a one liner and then loop over them and raise the error when one of them was invalid.

But that aside, this seems to work just fine! 

LGTM. Merge once green.

**Review by Theodlz — APPROVED** [review]
One small comment

**Pull request merged**